### PR TITLE
Snail mode

### PR DESCRIPTION
**Pull request by jlafayette**
Add a new "Snail Mode" map where snakes leave a trail of hazards behind them.

The stack of hazards left behind is based on the length of the snake.  Hazard-stack counts decrement each turn so the trails fade over time.

This uses a hacky approach of storing state between turns by using off-board hazards.  Hopefully this doesn't break too many snakes - also totally open to using a different approach if something else would work better.

This map does not generate a hazard when tail chasing since placing a hazard directly on the head of a snake doesn't damage it and looks confusing in the viewer.

### CODE REVIEW

**Comment by robbles** [comment]
Thanks for building and submitting this! Just tested it out locally and it looks awesome 🚀 
We're still discussing how best to handle the off-board hazard state, but I'll try to get back to you this week so this PR isn't left in limbo.

**Comment by robbles** [comment]
Hey @jlafayette and @coreyja - sorry about the delay on this one!
The current plan is to mark this map as experimental and merge it so that everyone can try it out, but we'll hold off on adding it to the custom game page and Battlegrounds until we figure out how we want to provide state storage to maps.

I just added some basic tagging support for maps. Would you mind adding the `TAG_EXPERIMENTAL` and `TAG_HAZARD_PLACEMENT` tags to this map before we merge it?

File: maps/snail_mode.go
Comment (robbles) [inline maps/snail_mode.go:65]: It's probably not worth changing, but you can actually use a `rules.Point` as a map key! I think a few other maps use this to look up tables of locations for a given WxH board.

Comment (jlafayette) [reply maps/snail_mode.go:65]: I didn't know that was possible to use a struct as a map key - thanks!

**Comment by coreyja** [comment]
@jlafayette Thanks for getting this updated! 💜

**Review by robbles — APPROVED** [review]
Thanks for the updates @jlafayette!

**Pull request merged**
